packages/synapse-react-client/src/components/SynapseTable/datasets/DatasetItemsEditor.tsx

```diff
@@ -199,30 +270,19 @@ export function DatasetItemsEditor(props: DatasetItemsEditorProps) {
     }
   }, [hasChangedSinceLastSave, onUnsavedChangesChange])
 
-  useEffect(() => {
-    if (
-      previousDatasetToUpdate &&
-      datasetToUpdate &&
-      !isEqual(previousDatasetToUpdate, datasetToUpdate)
-    ) {
-      const toastMessageTitle = getToastMessageTitle()
-      displayToast(SAVE_TO_CONTINUE, 'info', {
-        title: toastMessageTitle,
-        primaryButtonConfig: {
-          text: SAVE_CHANGES,
-          onClick: () => mutate(datasetToUpdate),
-        },
-      })
-    }
-    setPreviousDatasetToUpdate(datasetToUpdate)
-  }, [datasetToUpdate])
-
   // We get the project ID to show the "Current Project" context in the Entity Finder.
   const { data: path } = useGetEntityPath(entityId)
   const projectId = path?.path[1]?.id
 
   const { mutate, isPending: updateIsPending } =
     useUpdateEntity<EntityRefCollectionView>({
+      onMutate: () => {
+        // Close existing toast that prompted the user to save
+        if (onCloseLastSavePromptToast) {
+          onCloseLastSavePromptToast.close()
+          setOnCloseLastSavePromptToast(undefined)
+        }
+      },
```

Comment:
The user decided to save their changes, so hide a toast that pesters them to save